src/NServiceBus.RabbitMQ/Connection/ChannelProvider.cs

```diff
@@ -20,7 +22,12 @@ public ConfirmsAwareChannel GetPublishChannel()
             {
                 channel?.Dispose();
 
-                channel = new ConfirmsAwareChannel(connectionManager.GetPublishConnection(), usePublisherConfirms);
+                if (connection == null)
+                {
+                    CreateConnection();
```

Comment:
A minor point, but here I'd probably prefer that we inline

```C#
lock (connectionFactory)
{
    connection = connection ?? connectionFactory.CreatePublishConnection();
}
```

to remove the side effect from `CreateConnection`.